src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -517,6 +517,43 @@ private static void ArrayTypeCheck_Helper(object obj, void* elementType)
             }
         }
 
+        // Helpers for boxing
+        [DebuggerHidden]
+        internal static object? Box_Nullable(MethodTable* srcMT, ref byte nullableData)
+        {
+            Debug.Assert(srcMT->IsNullable);
+
+            // If 'hasValue' is false, return null.
+            if (!Unsafe.As<byte, bool>(ref nullableData))
+                return null;
+
+            // Allocate a new instance of the T in Nullable<T>.
+            MethodTable* dstMT = srcMT->InstantiationArg0();
+            ref byte srcValue = ref Unsafe.Add(ref nullableData, srcMT->NullableValueAddrOffset);
+
+            // Delegate to non-nullable boxing implementation
+            return Box(dstMT, ref srcValue);
+        }
+
+        [DebuggerHidden]
+        internal static object Box(MethodTable* typeMT, ref byte unboxedData)
+        {
+            Debug.Assert(typeMT != null);
+            Debug.Assert(typeMT->IsValueType);
+
+            object boxed = RuntimeTypeHandle.InternalAllocNoChecks(typeMT, out uint bytes);
```

Comment:
```suggestion
            object boxed = RuntimeTypeHandle.InternalAllocNoChecks(typeMT);
```

Reply:
Can you explain why? As discussed before, it saved a call and improved some throughput.

Reply:
I do not think it is worth the extra complexity. The microbenchmark that was used to measure the throughput is very artificial.

Reply:
It saves a round trip. Earlier you mentioned you are fine with the out arg https://github.com/dotnet/runtime/pull/115134#discussion_r2070363607.

Reply:
I tried to suggest that we do not need the `out` argument and that the perf-loss from the extra roundtrip is fine. I am sorry for not being clear.

Reply:
Sorry, somehow, I misread your comment wrong twice. I thought you were leaning towards keeping the out arg.